src/Functions/FunctionBinaryArithmetic.h

```diff
@@ -1259,4 +1268,167 @@ class FunctionBinaryArithmetic : public IFunction
+        const ColumnWithTypeAndName & left_, const ColumnWithTypeAndName & right_, const Context & context_)
+        : Base(context_), left(left_), right(right_)
+    {
+    }
+
+    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result, size_t input_rows_count) const override
+    {
+        if (left.column && isColumnConst(*left.column) && arguments.size() == 1)
+        {
+            Block block_with_constant
+                = {{left.column->cloneResized(input_rows_count), left.type, left.name},
+                   block.getByPosition(arguments[0]),
+                   block.getByPosition(result)};
+            Base::executeImpl(block_with_constant, {0, 1}, 2, input_rows_count);
+            block.getByPosition(result) = block_with_constant.getByPosition(2);
+        }
+        else if (right.column && isColumnConst(*right.column) && arguments.size() == 1)
+        {
+            Block block_with_constant
+                = {block.getByPosition(arguments[0]),
+                   {right.column->cloneResized(input_rows_count), right.type, right.name},
+                   block.getByPosition(result)};
+            Base::executeImpl(block_with_constant, {0, 1}, 2, input_rows_count);
+            block.getByPosition(result) = block_with_constant.getByPosition(2);
+        }
+        else
+            Base::executeImpl(block, arguments, result, input_rows_count);
+    }
+
+    bool hasInformationAboutMonotonicity() const override
+    {
+        std::string_view name_ = Name::name;
+        if (name_ == "minus" || name_ == "plus" || name_ == "multiply" || name_ == "divide" || name_ == "intDiv")
+        {
+            return true;
+        }
+        return false;
+    }
+
+    Monotonicity getMonotonicityForRange(const IDataType &, const Field & left_point, const Field & right_point) const override
+    {
+        std::string_view name_ = Name::name;
+        if (name_ == "minus" || name_ == "plus")
+        {
+            return {true, true, true};
+        }
+        if (name_ == "multiply" || name_ == "divide" || name_ == "intDiv")
+        {
+            if (!left.column)
+            {
+                bool positive = true;
+                if (WhichDataType(right.type).isInt())
+                {
+                    positive = right.column->getInt(0) >= 0;
+                }
+
+                if (WhichDataType(left.type).isUInt())
+                    return {true, positive, true};
+                else if (WhichDataType(left.type).isInt())
+                {
+                    if (left_point.get<Int64>() == right_point.get<Int64>())
+                        return {true, positive, true};
+                    if (left_point.get<Int64>() >= 0)
+                        return {true, positive, false};
+                    else if (right_point.get<Int64>() <= 0)
+                        return {true, !positive, false};
+                    else
+                        return {false, true, false};
```

Comment:
This code is difficult to read.
It's would be nice to add a comment (or even ascii schema) to each case.